tests/Integration/Adapter/ConfigurationTest.php

```diff
@@ -0,0 +1,415 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Integration\Adapter;
+
+use Doctrine\ORM\EntityManager;
+use Generator;
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShopBundle\Entity\Shop;
+use PrestaShopBundle\Entity\ShopGroup;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class ConfigurationTest extends KernelTestCase
+{
+    /**
+     * @var Configuration|null
+     */
+    private $configuration;
+
+    /**
+     * @var EntityManager
+     */
+    private $entityManager;
+
+    protected function setUp()
+    {
+        self::bootKernel();
+
+        $container = self::$kernel->getContainer();
+        $this->entityManager = $container->get('doctrine.orm.entity_manager');
+        $this->configuration = $container->get('prestashop.adapter.legacy.configuration');
+
+        $this->initMultistore();
+    }
+
+    /**
+     * @param array $setParams
+     * @param array $getParams
+     * @param $expectedResult
+     *
+     * @dataProvider getProvider
+     */
+    public function testGet(array $setParams, array $getParams, $expectedResult): void
+    {
+        $this->setAndGetValuesForTesting($setParams, $getParams, $expectedResult);
+    }
+
+    /**
+     * @param array $setParams
+     * @param array $getParams
+     * @param $expectedResult
+     *
+     * @dataProvider getWithStrictParameterProvider
+     */
+    public function testGetWithSrictParameter(array $setParams, array $getParams, $expectedResult)
+    {
+        $this->setAndGetValuesForTesting($setParams, $getParams, $expectedResult);
+    }
+
+    /**
+     * @param array $setParams
+     * @param array $getParams
+     * @param string $expectedResult
+     * @dataProvider hasProvider
+     */
+    public function testHas(array $setParams, array $getParams, string $expectedResult): void
+    {
+        if (!empty($setParams)) {
+            $this->configuration->set($setParams['key'], $setParams['value'], $setParams['shopConstraint']);
+        }
+        $result = $this->configuration->has($getParams['key'], $getParams['shopConstraint']);
+
+        $this->assertEquals($expectedResult, $result);
+    }
+
+    public function hasProvider(): Generator
```

Comment:
```suggestion
    public function hasProvider(): iterable
```

Just a small trick I got from @Progi1984 you can use `iterable` for providers relying on `yield` It's quite handy since you don't need a `use` statement unlike with `Generator`